src/dbg/debugger_commands.cpp

```diff
@@ -1997,6 +2002,41 @@ CMDRESULT cbDebugKillthread(int argc, char* argv[])
     return STATUS_ERROR;
 }
 
+CMDRESULT cbDebugCreatethread(int argc, char* argv[])
+{
+    if(argc < 2)
+        return STATUS_ERROR;
+    duint Entry = 0;
+    duint Argument = 0;
+    if(!valfromstring(argv[1], &Entry))
+        return STATUS_ERROR;
+    if(!MemIsCodePage(Entry, false))
+        return STATUS_ERROR;
+    if(argc > 2)
+    {
+        if(!valfromstring(argv[2], &Argument))
+            return STATUS_ERROR;
+    }
+    DWORD ThreadId = 0;
+    if(ThreaderCreateRemoteThread(Entry, true, reinterpret_cast<LPVOID>(Argument), &ThreadId) != 0)

```

Comment:
Alright, fair enough :)

On Thu, 1 Sep 2016 at 17:28 Torusrxxx <notifications@github.com> wrote:

> In src/dbg/debugger_commands.cpp
> <https://github.com/x64dbg/x64dbg/pull/1028#discussion_r77166684>:
>
> > +{
> > +    if(argc < 2)
> > +        return STATUS_ERROR;
> > +    duint Entry = 0;
> > +    duint Argument = 0;
> > +    if(!valfromstring(argv[1], &Entry))
> > +        return STATUS_ERROR;
> > +    if(!MemIsCodePage(Entry, false))
> > +        return STATUS_ERROR;
> > +    if(argc > 2)
> > +    {
> > +        if(!valfromstring(argv[2], &Argument))
> > +            return STATUS_ERROR;
> > +    }
> > +    DWORD ThreadId = 0;
> > +    if(ThreaderCreateRemoteThread(Entry, true, reinterpret_cast<LPVOID>(Argument), &ThreadId) != 0)
>
> We are already improving titanengine by writing an alternative of it. And
> since this code is shorter than CreateRemoteThread and provides a chance
> for the engine to recieve the information, why shouldn't we use it?
> ThreaderCreateRemoteThread should be very easy to add in gleebug.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/x64dbg/x64dbg/pull/1028/files/7beae8f0b7ad15bb06fe7a38c22e60ce773d4522#r77166684>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACWCmb-YE_Cau6ZhxNu0BS99u6Q9qWeRks5qlsolgaJpZM4JygIO>
> .
>